qiskit/algorithms/eigen_solvers/vqd.py

```diff
@@ -662,7 +661,7 @@ def get_energy_evaluation(
         step: int,
         operator: OperatorBase,
         return_expectation: bool = False,
-        prev_states: List[float] = [],
+        prev_states: List[float] =  None,
```

Comment:
If you are given `step` greater than 0 maybe it would be useful to check that prev_states is not None and the list length is at least of length step. If an error is raised here if there is a mismatch it should be more meaningful than having it fail with a null pointer or index out of bounds error.

Reply:
Makes sense. I'll plug that in.